src/main.cpp

```diff
@@ -1273,7 +1273,7 @@ void CBlockHeader::SetAuxPow(CAuxPow* pow)
 
 bool IsAuxPowVersion(int nVersion)
 {
-    return (nVersion == BLOCK_VERSION_AUXPOW_WITH_AUX || nVersion == BLOCK_VERSION_AUXPOW_WITHOUT_AUX);
+    return (nVersion & BLOCK_VERSION_AUXPOW) == BLOCK_VERSION_AUXPOW;
```

Comment:
this is insufficient because of the `ChainId` implementation. I get `SetBestChain: xx of last 100 blocks above version 2` on testnet again because of this